core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1376,9 +1378,9 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
 
         private void checkStorageTableNotRedirected(QualifiedObjectName source)
         {
-            QualifiedObjectName target = metadata.getRedirectedTableName(session, source);
-            if (!source.equals(target)) {
-                throw new TrinoException(NOT_SUPPORTED, format("Redirection of materialized view storage table '%s' to '%s' is not supported", source, target));
+            Optional<QualifiedObjectName> redirectedTableName = metadata.getRedirectionAwareTableHandle(session, source).getRedirectedTableName();
```

Comment:
This could be
```java
metadata.getRedirectionAwareTableHandle(session, source).getRedirectedTableName().ifPresent(name ->
        throw new TrinoException(NOT_SUPPORTED, format("Redirection of materialized view storage table '%s' to '%s' is not supported", source, name)));